src/main.ts

```diff
@@ -145,6 +145,46 @@ export async function runAction(): Promise<any> {
         )
       )
     }
+
+    core.debug("Processing pipeline report...")
+    if (result && !result["passed"]) {
+      core.setFailed(
```

Comment:
One problem with moving the logic above the uploading artifacts piece is that core.setFailed will stop the workflow. Therefore no artifacts will be uploaded.